dateparser/date.py

```diff
@@ -461,6 +468,22 @@ def date_strings():
                     if self._is_applicable_locale(locale, s):
                         yield locale
 
+        if self.detect_languages_function:
+            detected_languages = self.detect_languages_function(
+                text=date_string, confidence_threshold=self._settings.LANGUAGE_DETECTION_CONFIDENCE_THRESHOLD
+            )
+
+            if self.languages:
+                self.languages += map_languages(detected_languages)
```

Comment:
On [line](https://github.com/scrapinghub/dateparser/pull/932/files#diff-ed41fbffb2891f79c65e3dc690fadd0cdc7a9a9191b69388de73f6b578bd3dd1R343) it seems we are allowing list, set, and tuples. 
In case of set, `+=` will throw and error(`TypeError`) 
In case of tuple, `map_languages` has to give a tuple in return for `+=` to work

so this case should be handled if we want to support all three data types.

Reply:
Thanks, @kishan3 for now have changes `+=` to `.append()`, I have commented a solution below Pl suggest

Reply:
Thanks, I agree with changing it to list to avoid breaking changes.

Reply:
Thanks 👍🏻